media/server/main/source/MediaPipelineServerInternal.cpp

```diff
@@ -587,12 +588,41 @@ bool MediaPipelineServerInternal::haveDataInternal(MediaSourceStatus status, uin
         return true;
     }
     m_activeRequests->erase(needDataRequestId);
+
+    int &counter = m_noAvailableSamplesCounter[mediaSourceType];
+
     if (status != MediaSourceStatus::OK && status != MediaSourceStatus::EOS)
     {
-        RIALTO_SERVER_LOG_WARN("Data request for needDataRequestId: %u received with wrong status", needDataRequestId);
+        // Incrementing the counter allows us to track the occurrences where the status is other than OK or EOS. By
+        // limiting the logging of the warning message to a the number of NO_AVAILABLE_SAMPLES in a row, we prevent the
+        // warning from being repeatedly logged.
+        if (status == MediaSourceStatus::NO_AVAILABLE_SAMPLES)
+        {
+            counter++;
+            if (counter == kMaxSkippedNoAvailableSamples)
+            {
+                RIALTO_SERVER_LOG_INFO("NO_AVAILABLE_SAMPLES received in: %d consecutive responses", counter);
+                RIALTO_SERVER_LOG_WARN("Data request for needDataRequestId: %u received with wrong status",
+                                       needDataRequestId);
+                counter = 0;
+            }
+            else
+            {
+                RIALTO_SERVER_LOG_DEBUG("NO_AVAILABLE_SAMPLES received: %d consecutively", counter);
+            }
+        }
+        else
+        {
+            counter = 0;
```

Comment:
Add a Warn log if we are not NO_AVAILABLE_SAMPLES:
RIALTO_SERVER_LOG_WARN("Data request for needDataRequestId: %u received with wrong status",
                                       needDataRequestId);